pgvectorscale/src/util/tape.rs

```diff
@@ -52,3 +64,44 @@ impl<'a> Tape<'a> {
         std::mem::drop(self)
     }
 }
+
+#[cfg(any(test, feature = "pg_test"))]
+#[pgrx::pg_schema]
+mod tests {
```

Comment:
can we add a test of tape::restart when:
1) the last page isn't the right type but there are some.
2) there are no pages of the right type in the relation?
3) the tape restarts but there isn't enough room in the last page

Reply:
I added testing for all of those cases.